SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -105,11 +105,44 @@ void USpatialReceiver::LeaveCriticalSection()
 		{
 			OnEntityAddedDelegate.Broadcast(PendingAddEntity);
 		}
+		PendingAddComponents.RemoveAll([PendingAddEntity](const PendingAddComponentWrapper& Component) {return Component.EntityId == PendingAddEntity;});
 	}
 
 	for (Worker_AuthorityChangeOp& PendingAuthorityChange : PendingAuthorityChanges)
 	{
-		HandleActorAuthority(PendingAuthorityChange);
+		if (PendingAuthorityChange.authority != WORKER_AUTHORITY_AUTHORITATIVE)
```

Comment:
WORKER_AUTHORITY_AUTHORITY_LOSS_IMMINENT doesn't seem to be handled here anymore, is this necessary? 

Perhaps this could loop and RemoveSwap and the loop below could process all remaining ops without condition?

Reply:
I think the `WORKER_AUTHORITY_AUTHORITY_LOSS_IMMINENT` will fall into this processing loop, since they are not equal to `WORKER_AUTHORITY_AUTHORITATIVE`. That being said, yes, we do go over the entire list twice, while we could removeswap. Wonder which one is faster?

Reply:
Oops, yeah - misread that! Maybe ignore this then :D

Reply:
I was going to bring this up too. It's also easier to see what you are doing if you are explicit with which ops you want to process.

Are we sure you can never get an add and remove for the same entity in this list? If so then we need to preserve the ordering.

Ultimately this should all be replaced by spatial view so I don't think performance on this matters.